### ISSUE TITLE
The dreaded RTMP send error 104

### ISSUE THREAD

**Issue by bvacaliuc** [uvdl/camera-streamer#1]
When I started to interleave audio+video and send to an RTMP sink, I immediately ran into problems.  Basically after some random (short) amount of time, the pipeline would shutdown with:
```
ERROR                   rtmp :0:: WriteN, RTMP send error 104 (x bytes)
```

Apparently I was not alone:
 * https://lists.freedesktop.org/archives/gstreamer-devel/2013-May/040765.html
 * https://lists.freedesktop.org/archives/gstreamer-embedded/2015-May/001060.html
 * https://github.com/arut/nginx-rtmp-module/issues/1069
 * https://groups.google.com/d/msg/nginx-rtmp/mBC_pszCvYQ/T4mZrn96GQAJ *(for `ffmpeg` to an nginx-rtmp server)*
 * https://obsproject.com/forum/threads/rtmp-random-disconnects.58080/ *(`OBS` problems streaming to various services - no resolution)*

The threads that had solutions were not satisfying:
 * This thread for `OBS` suggested adjusting server timeout parameters: https://obsproject.com/forum/threads/rtmp-send-errors.65115/
 * This one said "... RTMP is very fragile compared to other protocols ...": https://obsproject.com/forum/threads/obs-disconnecting.68589/
 * This thread had working `ffmpeg` pipelines that suggested to reduce the framesize/bitrate: https://superuser.com/questions/1288451/ffmpeg-streaming-to-rtmp-at-30fps

In my case I am committed to RTMP working because it is a core protocol used for `video.mavnet.online`.  What I notice is that the flow control in RTMP strongly affects the `gstreamer` pipeline of upstream elements.  Individually, video *or* audio work just fine, but together there is suffering.

I'm happy to report that I have worked out a solution, and I'm opening this issue to attach a commit to it.

Basically, buffer management is required that takes into account the varying startup latency of the video (slow) and audio (fast) pipelines.  `gstreamer` already requires balanced latency (it tells you what the minmum required is, then you adjust your `queue` block with `max-size-time` parameter as appropriate.  For doing minimum latency work, this is on the order of 2-3 frames *(typically sub 100ms total)*.  Because the video pipeline takes several seconds to start, the audio branch of the pipeline has to have *at least* that much latency built-in.  Further, when pulling from the video pipeline it helps to add the `leaky=upstream` parameter to the `queue`.

**Comment by terry8204** [comment]
thank you, I will try it.

**Comment by bvacaliuc** [comment]
@terry8204, just saw this.  How did you get along?
